### PR TITLE
Search Provider: Add copy to clipboard action

### PR DESCRIPTION
**Pull request by rafaelmardojai**
Captura de pantalla de 2022-05-16 12-23-30

### CODE REVIEW

**Comment by rafaelmardojai** [comment]
The result format is a copy of how Calculator does it.

File: dialect/search_provider/search_provider.in [hunk 1]
```diff
@@ -141,24 +170,27 @@ class TranslateService(dbus.service.Object):
...
+            return error_id
 
     def is_live_enabled(self):
         return Settings.get().live_translation and Settings.get().sp_translation
 
     def _get_translator(self):
         backend = Settings.get().active_translator
         translator = TRANSLATORS[backend](
-            self._on_loaded,
+            callback=self._on_loaded,
```
Comment (rafaelmardojai) [inline dialect/search_provider/search_provider.in:193]: Btw, the SP was broken again bc of this 😅

Comment (fushinari) [reply dialect/search_provider/search_provider.in:193]: Aah, sorry, should have realized XD

**Review by fushinari — APPROVED** [review]

File: dialect/search_provider/search_provider.in [hunk 2]
```diff
@@ -64,54 +67,79 @@ class TranslateService(dbus.service.Object):
...
+            if translate_id in self.translations:
+                text = self.translations[translate_id]
+            return [{
+                'id': translate_id,
+                'name': text,
+            }]
+
+        elif (
+            len(ids) == 2 and translate_id in self.translations
+            and ids[1].startswith(CLIPBOARD_PREFIX)
```
Comment (fushinari) [inline dialect/search_provider/search_provider.in:104]: Wouldn't it be more robust to check `ids[1] == CLIPBOARD_PREFIX + ids[0]`?

Comment (rafaelmardojai) [reply dialect/search_provider/search_provider.in:104]: In theory yes, but in practice `GetResultMetas` is only called with ids of a previous `Get(Initial/Subsearch)ResultSet`. 

So also `len(ids) == 2` should be enough, but this is a small check to be sure we have a clipboard result.

Comment (fushinari) [reply dialect/search_provider/search_provider.in:104]: > but this is a small check to be sure we have a clipboard result.

Yes, in practice, there will never be a situation where anything beyond `len(ids)` is useful. But if we're going to check anyway, I'd rather it be as technically correct as possible.

Comment (fushinari) [reply dialect/search_provider/search_provider.in:104]: Other than this (which is basically a pointless nitpick that I have :laughing:), everything looks good.

**Review by fushinari — APPROVED** [review]

**Pull request merged**